src/main/java/net/sf/jabref/importer/fetcher/DOItoBibTeXFetcher.java

```diff
@@ -86,14 +85,18 @@ public JPanel getOptionsPanel() {
         return null;
     }
 
-    public BibEntry getEntryFromDOI(String doiStr, ParserResult parserResult) {
-        Objects.requireNonNull(parserResult);
+    public BibEntry getEntryFromDOI(String doiStr) {
+        return getEntryFromDOI(doiStr, null);
```

Comment:
I think the idea of the ParserResult variable is to communicate error messages. So users of this new overload method are not able to display meaningful error messages. Wouldn't it be better to change the corresponding places to something like:
````
ParserResult result = new ();
getEntryFromDOI(doi, result);
if(result.hasWarnings()) { displayThem() }
else { party() }
````